libvmi/driver/memory_cache.c

```diff
@@ -297,6 +314,15 @@ memory_cache_insert(
     }
 }
 
+void memory_cache_remove(
```

Comment:
I think there has to be 2 versions of these functions, one for with page cache and one without, it's just hard to see the #else at L285.  Maybe a block comment or something around there will help call it out?